x/sync/client_test.go

```diff
@@ -396,20 +399,22 @@ func sendChangeProofRequest(
 
 		serverResponseChan = make(chan []byte, 1)
 
-		// The client fetching a change proof.
-		client = NewClient(&ClientConfig{
-			NetworkClient: networkClient,
-			Metrics:       &mockMetrics{},
-			Log:           logging.NoLog{},
-		})
-
 		// The context used in client.GetChangeProof.
 		// Canceled after the first response is received because
 		// the client will keep sending requests until its context
 		// expires or it succeeds.
 		ctx, cancel = context.WithCancel(context.Background())
 	)
 
+	// The client fetching a change proof.
+	client, err := NewClient(&ClientConfig{
+		NetworkClient: networkClient,
+		Metrics:       &mockMetrics{},
+		Log:           logging.NoLog{},
+		BranchFactor:  merkledb.BranchFactor16,
+	})
+	require.NoError(err)
+
 	defer cancel() // avoid leaking a goroutine
```

Comment:
nit: I suppose this should come before `client, err := ...` in case it errors
applies elsewhere in this file as well